pallets/pools/src/lib.rs

```diff
@@ -913,66 +909,93 @@ pub mod pallet {
 					Error::<T>::WipedOut
 				);
 
-				// Redeem orders are denominated in tranche tokens, not in the pool currency.
-				// Convert redeem orders to the pool currency and return a list of (invest, redeem) pairs.
-				let orders =
-					Self::convert_orders_to_currency(&epoch_tranche_prices, &pool.tranches)
-						.ok_or(Error::<T>::Overflow)?;
+				if pool.tranches.acc_outstanding_investments()?.is_zero()
+					&& pool.tranches.acc_oustanding_redemptions()?.is_zero()
+				{
+					pool.tranches.combine_with_mut_non_residual_top(
+						epoch_tranche_prices.iter().enumerate(),
+						|tranche, (tranche_id, price)| {
+							let loc = TrancheLocator {
+								pool_id,
+								tranche_id: T::TrancheId::try_from(tranche_id)
+									.map_err(|_| Error::<T>::TrancheId)?,
+							};
+							Self::update_tranche_for_epoch(
+								loc,
+								submission_period_epoch,
+								tranche,
+								TrancheSolution {
+									invest_fulfillment: Perquintill::zero(),
+									redeem_fulfillment: Perquintill::zero(),
+								},
+								(Zero::zero(), Zero::zero()),
+								price.clone(),
+							)
+						},
+					)?;
 
-				let full_execution_solution = orders
-					.iter()
-					.map(|_| TrancheSolution {
-						invest_fulfillment: Perquintill::one(),
-						redeem_fulfillment: Perquintill::one(),
-					})
-					.collect::<Vec<_>>();
+					pool.start_epoch()?;
 
-				let tranche_supply = pool.tranches.supplies()?;
+					Self::deposit_event(Event::EpochExecuted(pool_id, submission_period_epoch));
 
-				let seniorities = pool
-					.tranches
-					.as_tranche_slice()
-					.iter()
-					.map(|tranche| tranche.seniority)
-					.collect::<Vec<_>>();
+					return Ok(Some(T::WeightInfo::close_epoch_no_investments(
```

Comment:
I think this name should be:

```suggestion
					return Ok(Some(T::WeightInfo::close_epoch_no_orders(
```